parser.go

```diff
@@ -405,18 +405,18 @@ func (p *parser) parseAttributeSelector() (attrSelector, error) {
 		return attrSelector{}, err
 	}
 
-	// check if the attribute contains an ignore case flag
-	ignoreCase := false
-	if p.s[p.i] == ' ' && p.s[p.i+1] == 'i' {
-		p.i += 2
-		ignoreCase = true
-	}
-
 	p.skipWhitespace()
 	if p.i >= len(p.s) {
 		return attrSelector{}, errors.New("unexpected EOF in attribute selector")
 	}
 
+	// check if the attribute contains an ignore case flag
+	ignoreCase := false
+	if strings.EqualFold(string(p.s[p.i]), "i") && p.s[p.i-1] == ' ' {
```

Comment:
I don't think it's necessary to check for the presence of a space in this condition. The space probably isn't actually required if the `i` comes after a quoted string. And a tab would be acceptable here.

`EqualFold` seems pretty heavyweight here. I think `p.s[p.i] == 'i' || p.s[p.i] == 'I'` would be clearer and faster.

Reply:
I added the check for the presence of a space because some users have attribute values without any quotes (see [Unquoted attribute values in CSS (and JavaScript) selectors](https://mathiasbynens.be/notes/unquoted-attribute-values#css).

If this library forces the use of double quotes, then I will remove this. I just removed EqualFold, thanks!

Reply:
If there is an unquoted attribute value with no space, the `i` will be interpreted as the last character of the attribute value. A space is required when leaving it out would be ambiguous. But we don't have to actually check for it; if the space is needed but missing, we'll never get to this point.

Reply:
Thanks for your explanations, I removed the space-check!